apps/gk_species.c

```diff
@@ -837,6 +869,7 @@ gk_species_new_dynamic(struct gkyl_gk *gk_app_inp, struct gkyl_gyrokinetic_app *
 
   // Allocate cflrate (scalar array).
   gks->cflrate = mkarr(app->use_gpu, 1, gks->local_ext.volume);
+  gks->cflrate_ho = mkarr(false, 1, gks->local_ext.volume);
```

Comment:
this should depend on whether the user requested the diagnostic, don't always allocate it

also, whenever you are allocating host (or device) twins of another array it is best to get the number of components and size from its twin, like
`gks->cflrate_ho = mkarr(false, gks->cflrate->ncomp, gks->cflrate->size);`